added docs for compare screenshots in Synthetics (#19361)
DataDog/documentation: content/en/synthetics/browser_tests/test_results.md

A test result is considered `FAILED` if it does not satisfy its assertions or if a step failed for another reason. You can troubleshoot failed runs by looking at their screenshots, checking for potential [errors](#errors-and-warnings) at the step level, and looking into [resources][17] and [backend traces](#backend-traces) generated by their steps.

> To help during the investigation, you can compare your failed result to the latest successful one by clicking on `Compare Screenshots`. It will showcase the two test execution side-by-side helping to spot any difference that could have caused the test to fail.

Reviewer: ```suggestion
To help during the investigation, click **Compare Screenshots** to receive a side-by-side screenshots of the failed result and the last successful execution. The comparison should help you to spot any difference that could have caused the test to fail.
```
